// @openshift-console/dynamic-plugin-sdk-internal/lib/api/internal-api.js
export const ActivityItem = require('@console/shared/src/components/dashboard/activity-card/ActivityItem')
    .default;
export const ActivityBody = require('@console/shared/src/components/dashboard/activity-card/ActivityBody')
    .default;
export const RecentEventsBody = require('@console/shared/src/components/dashboard/activity-card/ActivityBody')
    .RecentEventsBody;
export const OngoingActivityBody = require('@console/shared/src/components/dashboard/activity-card/ActivityBody')
    .OngoingActivityBody;
export const AlertsBody = require('@console/shared/src/components/dashboard/status-card/AlertsBody')
    .default;
export const AlertItem = require('@console/shared/src/components/dashboard/status-card/AlertItem')
    .default;
export const HealthItem = require('@console/shared/src/components/dashboard/status-card/HealthItem')
    .default;
export const HealthBody = require('@console/shared/src/components/dashboard/status-card/HealthBody')
    .default;
export const DashboardCard = require('@console/shared/src/components/dashboard/dashboard-card/DashboardCard')
    .default;
export const DashboardCardBody = require('@console/shared/src/components/dashboard/dashboard-card/DashboardCardBody')
    .default;
export const DashboardCardHeader = require('@console/shared/src/components/dashboard/dashboard-card/DashboardCardHeader')
    .default;
export const DashboardCardTitle = require('@console/shared/src/components/dashboard/dashboard-card/DashboardCardTitle')
    .default;
export const DashboardGrid = require('@console/shared/src/components/dashboard//DashboardGrid')
    .default;
export const ResourceInventoryItem = require('@console/shared/src/components/dashboard/inventory-card/InventoryItem')
    .ResourceInventoryItem;
export const DetailItem = require('@console/shared/src/components/dashboard/details-card/DetailItem')
    .default;
export const DetailsBody = require('@console/shared/src/components/dashboard/details-card/DetailsBody')
    .default;
export const UtilizationItem = require('@console/shared/src/components/dashboard/utilization-card/UtilizationItem')
    .default;
export const UtilizationBody = require('@console/shared/src/components/dashboard/utilization-card/UtilizationBody')
    .default;
export const useUtilizationDuration = require('@console/shared/src/hooks/useUtilizationDuration')
    .useUtilizationDuration;
export const UtilizationDurationDropdown = require('@console/shared/src/components/dashboard/utilization-card/UtilizationDurationDropdown')
    .UtilizationDurationDropdown;
export const usePrometheusPoll = require('@console/internal/components/graphs/prometheus-poll-hook')
    .usePrometheusPoll;
