// @openshift-console/dynamic-plugin-sdk-webpack/lib/extensions/dashboards.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
// Type guards
exports.isDashboardsTab = (e) => e.type === 'console.dashboards/tab';
exports.isDashboardsCard = (e) => e.type === 'console.dashboards/card';
exports.isDashboardsOverviewHealthPrometheusSubsystem = (e) => e.type === 'console.dashboards/overview/health/prometheus';
exports.isResolvedDashboardsOverviewHealthPrometheusSubsystem = (e) => e.type === 'console.dashboards/overview/health/prometheus';
exports.isDashboardsOverviewHealthURLSubsystem = (e) => e.type === 'console.dashboards/overview/health/url';
exports.isResolvedDashboardsOverviewHealthURLSubsystem = (e) => e.type === 'console.dashboards/overview/health/url';
exports.isDashboardsOverviewHealthResourceSubsystem = (e) => e.type === 'console.dashboards/overview/health/resource';
exports.isResolvedDashboardsOverviewHealthResourceSubsystem = (e) => e.type === 'console.dashboards/overview/health/resource';
exports.isDashboardsOverviewHealthOperator = (e) => e.type === 'console.dashboards/overview/health/operator';
exports.isResolvedDashboardsOverviewHealthOperator = (e) => e.type === 'console.dashboards/overview/health/operator';
exports.isDashboardsInventoryItemGroup = (e) => e.type === 'console.dashboards/overview/inventory/item/group';
exports.isDashboardsOverviewInventoryItem = (e) => e.type === 'console.dashboards/overview/inventory/item';
exports.isDashboardsOverviewInventoryItemReplacement = (e) => e.type === 'console.dashboards/overview/inventory/item/replacement';
exports.isProjectDashboardInventoryItem = (e) => e.type === 'console.dashboards/project/overview/item';
exports.isDashboardsOverviewResourceActivity = (e) => e.type === 'console.dashboards/overview/activity/resource';
exports.isDashboardsOverviewPrometheusActivity = (e) => e.type === 'console.dashboards/overview/prometheus/activity/resource';
exports.isDashboardsOverviewHealthSubsystem = (e) => exports.isDashboardsOverviewHealthURLSubsystem(e) ||
    exports.isDashboardsOverviewHealthPrometheusSubsystem(e) ||
    exports.isDashboardsOverviewHealthResourceSubsystem(e) ||
    exports.isDashboardsOverviewHealthOperator(e);
