// @openshift-console/dynamic-plugin-sdk-webpack/lib/extensions/topology-types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var TopologyViewType;
(function (TopologyViewType) {
    TopologyViewType["graph"] = "graph";
    TopologyViewType["list"] = "list";
})(TopologyViewType = exports.TopologyViewType || (exports.TopologyViewType = {}));
var TopologyDisplayFilterType;
(function (TopologyDisplayFilterType) {
    TopologyDisplayFilterType["show"] = "show";
    TopologyDisplayFilterType["expand"] = "expand";
    TopologyDisplayFilterType["kind"] = "kind";
})(TopologyDisplayFilterType = exports.TopologyDisplayFilterType || (exports.TopologyDisplayFilterType = {}));
var TopologyDecoratorQuadrant;
(function (TopologyDecoratorQuadrant) {
    TopologyDecoratorQuadrant["upperLeft"] = "upperLeft";
    TopologyDecoratorQuadrant["upperRight"] = "upperRight";
    TopologyDecoratorQuadrant["lowerLeft"] = "lowerLeft";
    TopologyDecoratorQuadrant["lowerRight"] = "lowerRight";
})(TopologyDecoratorQuadrant = exports.TopologyDecoratorQuadrant || (exports.TopologyDecoratorQuadrant = {}));
exports.SHOW_GROUPING_HINT_EVENT = 'show-regroup-hint';
